mlflow/data/spark_dataset.py

```diff
@@ -0,0 +1,384 @@
+        return table_provider.isDefined() and table_provider.get() == "delta"
+    except Exception:
+        return False
+
+
+def _is_delta_table_path(path: str) -> bool:
+    """
+    Checks if the specified filesystem path is a Delta table.
+
+    :return: True if the specified path is a Delta table. False otherwise.
+    """
+    if os.path.exists(path) and "_delta_log" in os.listdir(path):
+        return True
+
+    try:
+        dbfs_path = dbfs_hdfs_uri_to_fuse_path(path)
+        return os.path.exists(dbfs_path) and "_delta_log" in os.listdir(dbfs_path)
+    except Exception:
+        return False
+
+
+def _try_get_delta_table_latest_version_from_path(path: str) -> Optional[int]:
+    """
+    Gets the latest version of the Delta table located at the specified path.
+
+    :param path: The path to the Delta table.
+    :return: The version of the Delta table, or None if it cannot be resolved (e.g. because the
+             Delta core library is not installed or the specified path does not refer to a Delta
+             table).
+    """
+    try:
+        spark = SparkSession.builder.getOrCreate()
+        j_delta_table = spark._jvm.io.delta.tables.DeltaTable.forPath(spark._jsparkSession, path)
+        return _get_delta_table_latest_version(j_delta_table)
+    except Exception as e:
+        _logger.warning(
+            "Failed to obtain version information for Delta table at path '%s'. Version information"
+            " may not be included in the dataset source for MLflow Tracking. Exception: %s",
+            path,
+            e,
+        )
+
+
+def _try_get_delta_table_latest_version_from_table_name(table_name: str) -> Optional[int]:
+    """
+    Gets the latest version of the Delta table with the specified name.
+
+    :param table_name: The name of the Delta table.
+    :return: The version of the Delta table, or None if it cannot be resolved (e.g. because the
+             Delta core library is not installed or no such table exists).
+    """
+    try:
+        spark = SparkSession.builder.getOrCreate()
+        j_delta_table = spark._jvm.io.delta.tables.DeltaTable.forName(
+            spark._jsparkSession, table_name
+        )
+        return _get_delta_table_latest_version(j_delta_table)
+    except Exception as e:
+        _logger.warning(
+            "Failed to obtain version information for Delta table with name '%s'. Version"
+            " information may not be included in the dataset source for MLflow Tracking."
+            " Exception: %s",
+            table_name,
+            e,
+        )
+
+
+def _get_delta_table_latest_version(j_delta_table) -> int:
+    """
+    Obtains the latest version of the specified Delta table Java class.
+
+    :param delta_table: A Java DeltaTable class instance.
+    :return: The version of the Delta table.
+    """
+    latest_commit_jdf = j_delta_table.history(1)
+    latest_commit_row = latest_commit_jdf.head()
+    version_field_idx = latest_commit_row.fieldIndex("version")
```

Comment:
is this structure documented somewhere? just want to make sure that reading the version in this way is stable.

Reply:
confirmed that this is stable. `version` is part of the schema for the commit history of the table